packages/viewer/src/components/scene-tree/scene-tree.tsx

```diff
@@ -838,16 +841,18 @@ export class SceneTree {
   }
 
   @Listen('search')
-  protected handleSearch(event: CustomEvent<string>): void {
+  protected async handleSearch(event: CustomEvent<string>): Promise<void> {
     const columnsToSearch =
       this.metadataSearchKeys.length > 0
         ? this.metadataSearchKeys
         : this.metadataKeys;
 
-    this.filterItems(event.detail, {
+    this.isSearching = true;
```

Comment:
I'd argue for putting this field in `SceneTreeState` as well. This way the controller is the central place for managing this state. Right now, I bet there's a bug because someone can either call `SceneTree.filter()` or `SceneTreeController.filter()` and you won't get a loading spinner.